api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -917,6 +917,10 @@ message Cluster {
   // [#not-implemented-hide:]
   // Prefetch configuration for this cluster.
   PrefetchPolicy prefetch_policy = 50;
+
+  // If `connection_pool_per_downstream_connection` is true, the cluster will use a separate
+  // connection pool for every downstream connection
+  bool connection_pool_per_downstream_connection = 51;
```

Comment:
Are you planning on adding release notes when your PR series wraps, or is it valid to add them now?

Reply:
Now that you mention it, it probably makes sense to put the release notes in with this PR as the follow-up is fairly separate.